Tests/UnitTests/Diagnostics/DiagnosticsFileHandlerTests.swift

```diff
@@ -0,0 +1,210 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  DiagnosticsFileHandlerTests.swift
+//
+//  Created by Cesar de la Vega on 2/4/24.
+
+import Foundation
+import Nimble
+@testable import RevenueCat
+import XCTest
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class BaseDiagnosticsFileHandlerTests: TestCase {
+
+    fileprivate var fileHandler: FileHandler!
+    fileprivate var handler: DiagnosticsFileHandler!
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        try AvailabilityChecks.iOS13APIAvailableOrSkipTest()
+
+        self.fileHandler = try Self.createWithTemporaryFile()
+        self.handler = .init(self.fileHandler)
+    }
+
+    override func tearDown() async throws {
+        self.handler = nil
+
+        try await super.tearDown()
+    }
+
+}
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+class DiagnosticsFileHandlerTests: BaseDiagnosticsFileHandlerTests {
+
+    // MARK: - append
+
+    func testAppendEventWithProperties() async throws {
+        let content = DiagnosticsEvent(name: "HTTP_REQUEST_PERFORMED",
+                                       properties: ["key": AnyEncodable("value")],
+                                       timestamp: Date())
+
+        await self.handler.appendEvent(diagnosticsEvent: content)
+
+        let entries = await self.handler.getEntries()
+        let encodedContent = try content.encodeAndDecode()
+        expect(entries[0]).to(equal(encodedContent))
+    }
+
+    // MARK: - cleanSentDiagnostics
+
+    func testCleanSentDiagnostics() async throws {
+        await self.handler.appendEvent(diagnosticsEvent: Self.sampleEvent())
+        await self.handler.appendEvent(diagnosticsEvent: Self.sampleEvent())
+        await self.handler.appendEvent(diagnosticsEvent: Self.sampleEvent())
+        await self.handler.appendEvent(diagnosticsEvent: Self.sampleEvent())
+
+        var entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(4))
+
+        await self.handler.cleanSentDiagnostics(diagnosticsSentCount: 2)
+        entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(2))
+
+        await self.handler.cleanSentDiagnostics(diagnosticsSentCount: 2)
+        entries = await self.handler.getEntries()
+        expect(entries.count).to(equal(0))
+    }
+
+    // MARK: - getEntries
+
+    func testGetEntries() async throws {
+        let line1 =
+            "{\"properties\":{\"key\":\"value\"}," +
+            "\"timestamp\":\"2024-04-04T12:55:59Z\"," +
```

Comment:
We should try to sync up on this so we send the same values in iOS and Android... Should we send this in Android as well? I kinda like sending millis since epoch, though we lose timezone information, so I'm ok sending this if you prefer.

Reply:
This is what the default encoder encodes to. I am leaning more towards this since it's readable but not sure if we will ever need to read this directly

Reply:
I think that's ok, I can change it in the backend/Android 👍

Reply:
Thanks!